### PR TITLE
Models

### PR DESCRIPTION
**Pull request by CalyCherkaoui**
In this project, we build an exclusive clubhouse app where our users can write posts. Inside the clubhouse, users can see who the author of a post is but, outside, they can only see the story and wonder who wrote it.

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  <tbody>
+    <% @posts.each do |post| %>
+      <tr>
+        <td><%= post.body %></td>
+        <% if user_signed_in?%>
+          <td><%= post.user.name %></td>
+        <% end %>
+        <% if user_signed_in? && post.user_id == current_user.id %>
+          <td><%= link_to 'Destroy', post, method: :delete, data: { confirm: 'Are you sure?' } %></td>
+        <% end %>
```
Comment (maelfosso) [inline app/views/posts/index.html.erb]: That's what we called logic
Kindly move that logic into helpers.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -10,6 +10,23 @@
...
+      <%= link_to 'Log-in', new_user_session_path %>
+    <% end %>    
+
+    <% if flash[:notice] %>
+      <p class="notice"><%= notice %></p>
+    <% end %>
+    
+    <% if flash[:alert] %>
+      <p class="alert"><%= alert %></p>
+    <% end %>
```
Comment (maelfosso) [inline app/views/layouts/application.html.erb]: Move those logic from views to helpers.

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
+
+  # GET /posts
+  # GET /posts.json
+  def index
+    @posts = Post.all
+    @users = User.all
```
Comment (maelfosso) [inline app/controllers/posts_controller.rb]: Why this line? **@users** is not used in the views

**Review by maelfosso — APPROVED** [review]
# Changes required :red_circle: 

Hello,
Thank you for the job done so far.
However, there are some small issues to fix. 
Take a look at the comments left below. Once done, submit a new code review request.

Thanks

Mael FOSSO

File: app/views/posts/index.html.erb (hunk 1 above)
Comment (asiimwedismas) [reply app/views/posts/index.html.erb]: Could you explain a little bit how we are supposed to move this to helpers? 
I don't see any serious logic here. Just basic conditions to determine what gets displayed, No serious logic in my view.

Thanks

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, everything works fine, but there are a few more things required. 
### General
- Hello Team in order to move forward you need to work on requirement requested by previous TSE.
### [ OPTIONAL ]
You can use this article as a guide 
https://shehantips.com/10-01-2019/
https://rails.devcamp.com/trails/dissecting-rails-5/campsites/working-views-rails/guides/guide-view-helpers-rails-5

### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Comment by asiimwedismas** [comment]
@maelfosso @kobiyoyo  Thanks for the resources, much appreciated... We will update the project BUT, I believe this is one of those moments where you **shouldn't do something just because it can be done**.

In my view, logic is when you have to do some serious processing like 1+1 and then using the result to do some more processing(possibly calling another function with the result etc..)
The side effects of moving HTML to a ruby **ONLY**  file because of a simple **if block** outweigh the benefits if any.

Am writing this comment before I have started implementing the required changes and I can already see issues if it were a large project;

1. Moving HTML code to a ruby **ONLY** file means you will lose editor support features for HTML plugins. These are just a few lines of code so imagine if they were 50 HTML lines and u moved them to a ruby **ONLY** file just because of one "if end" code block. Try editing/modifying a bootstrap modal form inside a .rb file, you might as well use notepad.

2. Maintenance of a large project would get more frustrating due to bullet point #1 and the very many small chunks of code. Having to constantly switch between files. if you have 50 "if end" blocks in only one view, those are possibly 50 small chunks of code in at least one ruby **ONLY** file and when you get to those, you are faced with bullet #1. So what happens if your site has 50 pages?

Thanks for the material though, we shall implement the required changes as suggested. 

_Happy reviewing_

**Comment by CalyCherkaoui** [comment]
> ## Hi Team 
> #### Great Job so far, everything works fine, but there are a few more things required.
> ### General
> 
>     * Hello Team in order to move forward you need to work on requirement requested by previous TSE.
> 
> 
> ### [ OPTIONAL ]
> 
> You can use this article as a guide
> https://shehantips.com/10-01-2019/
> https://rails.devcamp.com/trails/dissecting-rails-5/campsites/working-views-rails/guides/guide-view-helpers-rails-5
> ### Make changes and submit another code review request.
> 
> _Happy coding_
> (Any questions Feel free to contact me on slack @kobiyoyo)

Issues solved.
Thanks for the Review

**Comment by Dandush03** [comment]
# Status: :rocket:  Project Approved :rocket: 

## Great Work implementing the requested changes

# Good luck in the next milestone

# Happy coding :nerd_face: :computer:

**Pull request merged**